File: app/src/main/java/com/hustunique/morii/database/SoundItemInfo.java
package com.hustunique.morii.database;

import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

import java.io.Serializable;

@Entity
public class SoundItemInfo implements Serializable {
    @PrimaryKey(autoGenerate = true)
    public long soundItemInfoId;
    public long diaryInfoId;
    public int soundItemId;
    public int soundItemPosition;

    public SoundItemInfo(long soundItemInfoId, long diaryInfoId, int soundItemId, int soundItemPosition) {
        this.soundItemInfoId = soundItemInfoId;
        this.diaryInfoId = diaryInfoId;
        this.soundItemId = soundItemId;
        this.soundItemPosition = soundItemPosition;
    }

    @Ignore
    public SoundItemInfo(long diaryInfoId, int soundItemPosition, int soundItemId) {
        this.diaryInfoId = diaryInfoId;
        this.soundItemId = soundItemId;
        this.soundItemPosition = soundItemPosition;
    }

    @Ignore
    public SoundItemInfo(int soundItemPosition, int soundItemId) {
        this.soundItemId = soundItemId;
        this.soundItemPosition = soundItemPosition;
    }
}

File: app/src/main/java/com/hustunique/morii/database/SoundItemInfo.kt
package com.hustunique.morii.database

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import java.io.Serializable

@Entity
data class SoundItemInfo(
    @PrimaryKey(autoGenerate = true)
    val soundItemInfoId: Long,
    var diaryInfoId: Long,
    val soundItemId: Int,
    val soundItemPosition: Int
) : Serializable {

    @Ignore
    constructor(soundItemPosition: Int, soundItemId: Int) : this(
        0,
        0,
        soundItemId,
        soundItemPosition
    ) {
    }
}